src/nnet3/nnet-utils.cc

```diff
@@ -40,6 +40,120 @@ int32 NumInputNodes(const Nnet &nnet) {
   return ans;
 }
 
+int32 GetTimePeriodForIvector(const Nnet &nnet) {
+  std::vector<std::string> name;
+  name.push_back("ivector");
+  std::vector<int32> modulus;
+  // initialize with 0, so that if modulus is not found this function will
+  // return 0
+  modulus.resize(name.size(), 0);
+  for (int32 i = 0; i < nnet.NumNodes(); i++) {
+    const NetworkNode &node = nnet.GetNode(i);
+    if (node.node_type == kDescriptor) {
+      for (int32 p = 0; p < node.descriptor.NumParts(); p++) {
+        const SumDescriptor &this_part = node.descriptor.Part(p);
+        ExtractDescriptorValuesFromSumDescriptor(nnet, this_part, name,
+                                                 &modulus);
+      }
+    }
+  }
+  return modulus[0]; 
```

Comment:
Say that since you requested the values of the node name "ivector", the variable modulus can just have one value. Hence you return the 0th element. 

What will happen if I use multiple round descriptors on the ivector node in my config ? how will you handle this case. You can just raise an exception when this happens.